examples/regression/4-tutorials/plot_ccp_tutorial.py

```diff
@@ -0,0 +1,509 @@
+    nrows = int(np.ceil(len(titles) / ncols))
+    ax_need_transform = np.zeros((nrows, ncols))
+    if show_transform:
+        for i, mapie in enumerate(mapies):
+            ax_need_transform[i//ncols, i % ncols] = need_transform(mapie)
+            row_need_transform = np.max(ax_need_transform, axis=1)
+        height_ratio = np.array([
+            item for x in row_need_transform
+            for item in ([3] if x == 0 else [3, 1])
+        ])
+        fig, axes = plt.subplots(nrows=nrows + int(sum(row_need_transform)),
+                                 ncols=ncols, figsize=(ncols*4, nrows*5),
+                                 height_ratios=height_ratio)
+
+        for ax in axes[np.where(height_ratio == 1)[0]-1, :].flatten():
+            ax.tick_params(axis='x', which='both', bottom=False,
+                           top=False, labelbottom=False)
+
+        transform_axes = np.full((nrows, ncols), None)
+        transform_axes[row_need_transform == 1, :] = axes[height_ratio == 1, :]
+        transform_axes = transform_axes.flatten()
+        main_axes = axes[height_ratio == 3, :].flatten()
+    else:
+        fig, axes = plt.subplots(nrows=nrows, ncols=ncols,
+                                 figsize=(ncols*4, nrows*4))
+        main_axes = axes.flatten()
+        transform_axes = np.full(main_axes.shape, None)
+
+    for i, (m_ax, t_ax, mapie, y_pred, y_pi, title) in enumerate(
+        zip(main_axes, transform_axes, mapies, y_preds, y_pis, titles)
+    ):
+        lower_bound = y_pi[:, 0, 0]
+        upper_bound = y_pi[:, 1, 0]
+
+        plot_subplot(
+            m_ax, X_test, y_test, mapie, y_pred, upper_bound, lower_bound,
+            cp[i], show_transform=ax_need_transform.flatten()[i],
+            ax_transform=t_ax
+        )
+        m_ax.set_title(title)
+        if i % 3 == 0:
+            m_ax.set_ylabel('Y')
+        if t_ax is not None:
+            t_ax.set_title("Transformation")
+            if i >= len(titles) - ncols:
+                t_ax.set_xlabel('X')
+        else:
+            m_ax.set_xlabel('X')
+        m_ax.legend()
+
+    fig.tight_layout()
+    plt.show()
+
+
+def plot_widths(titles, y_pis):
+    sort_order = np.argsort(X_test[:, 0])
+    cp = plt.get_cmap('tab10').colors
+    plt.figure(figsize=(8, 6))
+    for i, (title, pi) in enumerate(zip(titles, y_pis)):
+        plt.plot(X_test[sort_order, 0],
+                 (pi[sort_order, 1, 0] - pi[sort_order, 0, 0]),
+                 lw=2, color=mcolors.rgb2hex(cp[i]), label=title)
+
+    plt.title("Prediction interval width")
+    plt.xlabel("X")
+    plt.ylabel("Width")
+    plt.legend(fontsize=14)
+    plt.tight_layout()
+    plt.show()
+
+
+##############################################################################
+# 5. Experiments:
+# --------------------------------------------------------------------------
+
+##############################################################################
+# 5.1. Default :class:`~mapie.calibrators.GaussianCCP`:
```

Comment:
Note that not all subplots are used: 
<img width="745" alt="Screenshot 2024-06-18 at 11 23 45" src="https://github.com/scikit-learn-contrib/MAPIE/assets/63096108/0b51d467-b265-4bd2-bddb-31ec0823f18f">

Reply:
Done